go/exocom/src/exocom/exocom.go

```diff
@@ -106,3 +125,71 @@ func parseRegisterMessagePayload(message structs.Message) (string, error) {
 	}
 	return "", fmt.Errorf("Invalid register message payload: %v", message.Payload)
 }
+
+func (e *ExoCom) send(message structs.Message) error {
+	publicMessageName := messageTranslator.GetPublicMessageName(&messageTranslator.GetPublicMessageNameOptions{
+		Namespace:           e.clientRegistry.Clients[message.Sender].InternalNamespace,
+		ClientName:          message.Sender,
+		InternalMessageName: message.Name,
+	})
+	message.OriginalName = message.Name
```

Comment:
I think we can get by without setting the OriginalName as a property on the message and just passing it a variable to `sendToServices`